app/plugin_requests.go

```diff
@@ -67,19 +67,30 @@ func (a *App) servePluginRequest(w http.ResponseWriter, r *http.Request, handler
 		session, err := a.GetSession(token)
 		csrfCheckPassed := true
 
-		if err == nil && cookieAuth && r.Method != "GET" && r.Header.Get(model.HEADER_REQUESTED_WITH) != model.HEADER_REQUESTED_WITH_XML {
-			bodyBytes, _ := ioutil.ReadAll(r.Body)
-			r.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
-			r.ParseForm()
-			sentToken := r.FormValue("csrf")
+		if err == nil && cookieAuth && r.Method != "GET" {
```

Comment:
Is it possible to invert the logic to be default failure? So start with `csrfCheckPassed = false`.

Reply:
Yeah, good call. Fixed.